addons/mail/static/src/messaging/entity/entity/entity.js

```diff
@@ -0,0 +1,824 @@
+        /**
+         * @static
+         * @param {Object} data
+         */
+        static update(data) {
+            this.__update(this, data);
+        }
+
+        delete() {
+            this.constructor.__delete(this);
+        }
+
+        /**
+         * @returns {boolean}
+         */
+        get isEntity() {
+            return true;
+        }
+
+        /**
+         * @returns {boolean}
+         */
+        get isEntityClass() {
+            return false;
+        }
+
+        /**
+         * @returns {boolean}
+         */
+        get isEntityInstance() {
+            return true;
+        }
+
+        /**
+         * @param {Object} data
+         */
+        link(data) {
+            this.constructor.__link(this, data);
+        }
+
+        /**
+         * @param {Object} data
+         */
+        unlink(data) {
+            this.constructor.__unlink(this, data);
+        }
+
+        /**
+         * @param {Object} data
+         */
+        update(data) {
+            this.constructor.__update(this, data);
+        }
+
+        //--------------------------------------------------------------------------
+        // Private
+        //--------------------------------------------------------------------------
+
+        /**
+         * @static
+         * @private
+         * @returns {string[]}
+         */
+        static _getListOfClassAttributeNames() {
+            return [];
+        }
+
+        /**
+         * @static
+         * @private
+         * @abstract
+         * @param {Object} data
+         */
+        static _update(data) {}
+
+        /**
+         * Any assignment to class as attribute or stored relation should pass
+         * through this function.
+         *
+         * @static
+         * @private
+         * @param {Object} data
+         */
+        static _write(data) {
+            Object.assign(this.observable, data);
+        }
+
+        /**
+         * @private
+         * @param {Object} data
+         * @returns {string}
+         */
+        _createInstanceLocalId(data) {
+            return _.uniqueId(`${this.constructor.localId}_`);
+        }
+
+        /**
+         * @abstract
+         * @private
+         * @param {Object} data
+         */
+        _update(data) {}
+
+        /**
+         * Any assignment to instance as attribute or stored relation should
+         * pass through this function.
+         *
+         * @private
+         * @param {Object} data
+         */
+        _write(data) {
+            Object.assign(this, data);
+        }
+
+        //--------------------------------------------------------------------------
+        // Internal
+        //--------------------------------------------------------------------------
+
+        /**
+         * @static
+         * @param {mail.messaging.entity.Entity} entity instance or class
+         */
+        static __delete(entity) {
+            const { state } = this.env.store;
+            if (!this.get(entity)) {
```

Comment:
I think there should be an existence check instead, split from the get from localId